src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CachedCaseUserRepository.java

```diff
@@ -49,4 +51,9 @@ public void revokeAccess(Long caseId, String userId, String caseRole) {
     public List<CaseUserEntity> findCaseUserRoles(List<Long> caseIds, List<String> userIds) {
         return caseUserRepository.findCaseUserRoles(caseIds, userIds);
     }
+
+    @Override
+    public Set<String> getCaseUserRolesByUserId(String userId) {
+        return caseUserRepository.getCaseUserRolesByUserId(userId);
```

Comment:
Can you check during single search operation how many times it is hitting this method? We might need some request caching as well like others..